osu.Game/Rulesets/Objects/HitObjectLifetimeEntry.cs

```diff
@@ -26,6 +27,8 @@ public class HitObjectLifetimeEntry : LifetimeEntry
 
         private readonly IBindable<double> startTimeBindable = new BindableDouble();
 
+        internal event Action? RevertResult;
```

Comment:
This is a touch awkward in terms of event flow, but I can't immediately see a way out of it.